subsys/net/lib/mcumgr_smp_client/Kconfig

```diff
@@ -0,0 +1,35 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+menuconfig NRF_MCUMGR_SMP_CLIENT
+	bool "MCUMGR SMP client library"
+	select BASE64
```

Comment:
In general only prompt-less options should be `select`-ed due to its two way dependency semantics. 
https://docs.zephyrproject.org/latest/build/kconfig/tips.html#select-pitfalls
https://docs.zephyrproject.org/latest/build/kconfig/tips.html#alternatives-to-select

Reply:
I got a your point.  I was thinking a use depends on but then user need to enable long list of kconfig. 
Lwm2m utlis library use similar method that high level component kconfig enable all needed configs.